src/content/blog/first-four-val-town-runtimes.mdx

```diff
@@ -0,0 +1,247 @@
+---
+title: The first four Val Town runtimes
+generated: 1701894028894
+description: From vm to vm2 to workers to processes, the long journey to how we run vals today
+pubDate: February 5, 2024
+heroImage: ./first-four-val-town-runtimes/timeline.png
+author: Tom MacWright
+---
+
+import { Image } from "astro:assets";
+import memoryImage from "./first-four-val-town-runtimes/memory.png";
+
+One of the most important things that Val Town does is **run untrusted code**.
+Our whole premise requires us to take the TypeScript code that our community
+writes and run it on a server, safely, quickly, and flexibly.
+
+#### Safety
+
+The default way to run arbitrary code is "dangerously". Modules in Node.js have unfettered
+access to your system's environment variables, filesystem, network, and much
+more. This access is [routinely abused in supply-chain attacks](https://socket.dev/blog/inside-node-modules).
+The easiest way to run an arbitrary string of code is with the `eval()` method,
+which doesn't provide any kind of sandbox protections, so it can easily become
+an attack vector.
```

Comment:
I understand the ideas being laid out in this paragraph, but they do feel disconnected

Reply:
Agreed, could use more of a connection to what's above